### PR TITLE
Improve naming consistency and additional Gremlin docs [skip ci]

### LINKED ISSUE #575: docs naming consistency
Clean up some naming/wording inconsistencies in the documentation, things like:

Tinkerpop -> Apache TinkerPop
Cassandra -> Apache Cassandra
HBase -> Apache HBase
BerkeleyDB -> Oracle Berkeley DB Java Edition
ElasticSearch -> Elasticsearch
Hadoop -> Apache Hadoop
Groovy -> Apache Groovy
Console -> Gremlin Console
gremlin shell -> Gremlin Console
gremlin server -> Gremlin Server

Apache Lucene and Apache Solr were handled with #574 

For the Apache packages, it isn't necessary that Apache is always attached with the project name, but it is good practice to have it there on the first appearance.

### LINKED ISSUE #576: additional Gremlin documentation
Iterating the traversal is one of the most common problems for new users.

Provide links to additional documentation resources, language drivers, and language variants.

Also, there are few example Gremlin calls that need to be moved to TP 3.x syntax, like

graph.shutdown() -> graph.close()
graph.commit() -> graph.tx().commit()

### PR DESCRIPTION
**Pull request by pluradj**
Fixes #575
Fixes #576

Signed-off-by: Jason Plurad <pluradj@us.ibm.com>

Thank you for contributing to JanusGraph.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [X] Is there an issue associated with this PR? Is it referenced in the commit message?
- [X] Does your PR body contain #xyz where xyz is the issue number you are trying to resolve?
- [X] Has your PR been rebased against the latest commit within the target branch (typically `master`)?
- [X] Is your initial contribution a single, squashed commit?

### For code changes:
- [ ] Have you written and/or updated unit tests to verify your changes?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)? 
- [ ] If applicable, have you updated the LICENSE.txt file, including the main LICENSE.txt file in the root of this repository?
- [ ] If applicable, have you updated the NOTICE.txt file, including the main NOTICE.txt file found in the root of this repository?

### For documentation related changes:
- [X] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check Travis CI for build issues and submit an update to your PR as soon as possible.

Preview docs available at https://pluradj.github.io/janusgraph.org/latest/

### CODE REVIEW

File: docs/basics.txt [hunk 1]
```diff
@@ -556,7 +585,7 @@ IMPORTANT: Do not use bin/janusgraph.sh.  That starts the default configuration,
 The default configuration described in <<server-getting-started>> is a WebSocket configuration.  If you want to alter the default configuration in order to use JanusGraph Server as a HTTP endpoint for your JanusGraph database, follow these steps:
 
 .To Configure JanusGraph Server For Rest
```
Comment (sjudeng) [inline docs/basics.txt]: You didn't change this line but should it be "for REST"?

File: docs/basics.txt [hunk 2]
```diff
@@ -1060,7 +1089,7 @@ juno.property("name", "juno")
...
+In this example, a local JanusGraph graph database is opened. Adding the vertex "juno" is the first operation (in this thread) which automatically opens a new transaction. All subsequent operations occur in the context of that same transaction until the transaction is explicitly stopped or the graph database is `close()`.  If transactions are still open when `close()` is called, then the behavior of the outstanding transactions is technically undefined.  In practice, any non-thread-bound transactions will usually be effectively rolled back, but the thread-bound transaction belonging to the thread that invoked shutdown will first be committed. Note, that both read and write operations occur within the context of a transaction.
```
Comment (sjudeng) [inline docs/basics.txt]: Replace "or the graph database is `close()`" with "or the graph database is closed".

File: docs/basics.txt [hunk 3]
```diff
@@ -1169,7 +1198,8 @@ graph.tx().commit() // Can't fail due to uniqueName write lock contention involv
 
 Transactions are started automatically with the first operation executed against the graph. One does NOT have to start a transaction manually. The method `newTransaction` is used to start <<multi-thread-tx, multi-threaded transactions>> only.
 
-Transactions are automatically started under the TinkerPop semantics but *not* automatically terminated. Transactions have to be terminated manually with `g.commit()` if successful or `g.rollback()` if not. Manual termination of transactions is necessary because only the user knows the transactional boundary.
+Transactions are automatically started under the TinkerPop semantics but *not* automatically terminated. Transactions have to be terminated manually with `commit()` if successful or `rollback()` if not. Manual termination of transactions is necessary because only the user knows the transactional boundary.
```
Comment (sjudeng) [inline docs/basics.txt]: Just "terminated manually with `commit()` or `rollback()`"? Successful workflows might call "rollback()" and unsuccessful ones may call "commit()" (and "rollback()").

File: docs/intro.txt [hunk 4]
```diff
@@ -64,7 +64,7 @@ on incomplete data). This decision should be based on business
 requirements.
 _____________________
 
-When using a database, the http://en.wikipedia.org/wiki/CAP_theorem[CAP theorem] should be thoroughly considered (C=Consistency, A=Availability, P=Partitionability). JanusGraph is distributed with 3 supporting backends: http://cassandra.apache.org/[Cassandra], http://hbase.apache.org/[HBase], and http://www.oracle.com/technetwork/products/berkeleydb/[BerkeleyDB]. Note that BerkeleyDB is a non-distributed database and as such, is typically only used with JanusGraph for testing and exploration purposes.
+When using a database, the http://en.wikipedia.org/wiki/CAP_theorem[CAP theorem] should be thoroughly considered (C=Consistency, A=Availability, P=Partitionability). JanusGraph is distributed with 3 supporting backends: http://cassandra.apache.org/[Apache Cassandra], http://hbase.apache.org/[Apache HBase], and http://www.oracle.com/technetwork/database/berkeleydb/overview/index-093405.html[Oracle Berkeley DB Java Edition]. Note that BerkeleyDB JE is a non-distributed database and as such, is typically only used with JanusGraph for testing and exploration purposes.
```
Comment (sjudeng) [inline docs/intro.txt]: If using commas "and, as such, is typically" (or else remove commas)?

File: docs/intro.txt [hunk 5]
```diff
@@ -159,7 +157,7 @@ gremlin> [name:'aurelius', vocation:['philosopher', 'emperor']]
 ==>vocation=[philosopher, emperor]
 
 [TIP]
-Refer to http://tinkerpop.apache.org/docs/$MAVEN{tinkerpop.version}/reference[Tinkerpop documentation], http://gremlindocs.com/[GremlinDocs] and http://sql2gremlin.com/[SQL2Gremlin] for more information about using Gremlin.
+Refer to http://tinkerpop.apache.org/docs/$MAVEN{tinkerpop.version}/reference[Apache TinkerPop documentation], http://sql2gremlin.com/[SQL2Gremlin], and http://tinkerpop.apache.org/docs/$MAVEN{tinkerpop.version}/recipes/[Gremlin Recipes] for more information about using Gremlin.
```
Comment (sjudeng) [inline docs/intro.txt]: Put "documentation" in "Apache TinkerPop documentation" outside the link?

File: docs/basics.txt [hunk 6]
```diff
@@ -333,16 +333,20 @@ However, note that this would not affect vertices, edges, or properties previous
 [[gremlin]]
 == Gremlin Query Language
 
-image:http://tinkerpop.apache.org/docs/{tinkerpop_version}/images/gremlin-logo.png[link="http://tinkerpop.apache.org/docs/{tinkerpop_version}/reference"]
+image:http://tinkerpop.apache.org/docs/{tinkerpop_version}/images/gremlin-logo.png[width=370,height=143,link="http://tinkerpop.apache.org/docs/{tinkerpop_version}/reference"]
```
Comment (sjudeng) [inline docs/basics.txt]: This image isn't linkable in the generated docs but it doesn't look like it is originally either.

File: docs/basics.txt (hunk 1 above)
Comment (robertdale) [reply docs/basics.txt]: It should be 'for HTTP'. There is no REST interface.

Comment (pluradj) [reply docs/basics.txt]: @robertdale is always on HTTP vs REST patrol duty :)

**Pull request merged**